utils/utils.go

```diff
@@ -485,6 +509,32 @@ func GetSSHConfig() *ssh.ServerConfig {
 	return sshConfig
 }
 
+// checkAuthenticationKeyRequest makes an HTTP POST request to the specified url with
+// the provided ssh public key in OpenSSH 'authorized keys' format to validate
+// whether it should be accepted.
+func checkAuthenticationKeyRequest(authUrl string, authKey []byte) (bool, error) {
+	parsedUrl, err := url.ParseRequestURI(authUrl)
+	if err != nil || !parsedUrl.IsAbs() {
```

Comment:
Out of curiosity, is this check strictly necessary?

Reply:
It's not necessary, I think I added it to avoid ambiguity but it'll likely cause more confusion if someone omits the scheme. Will remove